gradient_boosting.py

```diff
@@ -0,0 +1,101 @@
+import numpy as np
+
+from sklearn.base import RegressorMixin
+from tree import DecisionTreeRegressor
+
+
+class BaseGradientBoostingRegressor:
+
+    def __init__(self,
+                 n_estimators: 5,
+                 max_depth: int,
+                 learning_rate=None):
+
+        self.estimator = DecisionTreeRegressor
+        self.n_estimators = n_estimators
+        self.max_depth = max_depth
+        self.learning_rate = [learning_rate] * n_estimators
+
+        self._estimator_params = self.estimator().get_params()
```

Comment:
Here, you get default parameters of `DecisionTreeRegressor` instead of taking the parameters which are passed via the constructor.

Reply:
Yes, but why is it wrong? We are working with that estimator and here we get his parameters.

Reply:
Tell me, where parameter `n_estimators` is used? Except for line 17.